apps/backend/src/users/users.service.ts

```diff
@@ -21,4 +25,55 @@ export class UsersService {
   remove(id: number) {
     return `This action removes a #${id} user`;
   }
+
+  async addFavoriteDrink(userId: string, favoriteDrinkId: string): Promise<User> {
+    const user = await this.prisma.user.findUnique({
+      where: { authSchId: userId },
+      include: { favouriteDrinks: true },
+    });
+
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    if (user.favouriteDrinks.length >= 3) {
+      throw new BadRequestException('A user can have only 3 favorite drinks.');
+    }
+
+    if (user.favouriteDrinks.some((drink) => drink.id === favoriteDrinkId)) {
+      throw new BadRequestException('The drink is already in the user list of favorite drinks.');
+    }
+
+    const updatedUser = await this.prisma.user.update({
+      where: { authSchId: userId },
+      data: { favouriteDrinks: { connect: { id: favoriteDrinkId } } },
+      include: { favouriteDrinks: true },
```

Comment:
we rarely use the result of a post/patch/delete call, so the include is probably not necessary. But it doesn't hurt much, so we could keep it. But in that case, we'll need a DTO for this, but that would extend the User entity that doesn't exist yet, so that can be done later